### PR TITLE
WG serving proposal

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
**Which issue(s) this PR fixes**:

See the thread for context: https://groups.google.com/g/kubernetes-sig-architecture/c/-DvFBA0805k

/sig apps
/sig architecture
/sig node
/sig scheduling
/sig autoscaling
/sig network
/sig instrumentation
/sig storage

/cc @smarterclayton

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
++

File: OWNERS_ALIASES
Comment (ArangoGutierrez) [inline OWNERS_ALIASES:146]: How would the volunteering process be, I am more than happy to help here

File: wg-serving/charter.md
Comment (ArangoGutierrez) [inline wg-serving/charter.md:33]: Training?

Comment (SergeyKanzhelev) [reply wg-serving/charter.md:33]: this refers to batch jobs that will run a batch of inference queries nighty or some similar scenario. So basically something the has characteristics similar to training. Does it make sense or some better language here will help?

File: sigs.yaml
Comment (SergeyKanzhelev) [inline sigs.yaml:3508]: @kubernetes/steering-committee what is the right way to get a liason for the WG?

Comment (mrbobbytables) [reply sigs.yaml:3508]: steering will discuss and assign once the charter is approved and merged

Comment (SergeyKanzhelev) [reply sigs.yaml:3508]: so keep it as TODO for now?

File: wg-serving/charter.md
Comment (alculquicondor) [reply wg-serving/charter.md:33]: I guess it might still be worth adding batch training to the list of "out of scope", for additional clarity.

Comment (smarterclayton) [reply wg-serving/charter.md:33]: Agreed, let's say "training and batch inference are under WG-Batch".

Continuous training may eventually be relevant, but we don't have to call it out here.

File: OWNERS_ALIASES
Comment (SergeyKanzhelev) [reply OWNERS_ALIASES:146]: we can discuss at the first meeting. Did you have a chance to reply here: https://forms.gle/27mqbTC1xBP5QPSV9?

File: sig-list.md
Comment (helayoty) [inline sig-list.md:70]: Is it PM or AM?

File: wg-serving/charter.md
Comment (smarterclayton) [inline wg-serving/charter.md:34]: I don't know what this means, or if I agree.  I think we should be focused on CRDs that make inference work consistently on Kube and enable it.  What's the history on this one?

File: sig-list.md
Comment (kerthcet) [reply sig-list.md:70]: Kindly ask if we can make this time more friendly for APAC person, would like to join the meeting but it's 12 o'clock at midnight, maybe a hour or two hours earlier? 

I know it's always hard to pick the time slot.

File: sigs.yaml
Comment (ArangoGutierrez) [inline sigs.yaml:3495]:
```suggestion
    - github: ArangoGutierrex
      name: Eduardo Arango
      company: NVIDIA
```

File: OWNERS_ALIASES
Comment (ArangoGutierrez) [inline OWNERS_ALIASES:146]:
```suggestion
    - ArangoGutierrez
```

File: wg-serving/README.md
Comment (ArangoGutierrez) [inline wg-serving/README.md:34]:
```suggestion
* Eduardo Atango (**[@ArangoGutierrez ](https://github.com/ArangoGutierrez)**), NVIDIA
```

**Review by ArangoGutierrez — COMMENTED** [review]
cc @SergeyKanzhelev @smarterclayton @klueska

File: wg-serving/charter.md
Comment (ArangoGutierrez) [inline wg-serving/charter.md:11]: We should add @smarterclayton comments during the first call, on what's a Serving workload here in the charter.

File: sigs.yaml
Comment (BenTheElder) [reply sigs.yaml:3511]: Yeah just leave it TODO and we'll discuss in steering and update these after merge

We should probably do a round of re-balancing liasons as well.

File: sig-list.md
Comment (kerthcet) [reply sig-list.md]: second ping

File: wg-serving/charter.md
Comment (soltysh) [reply wg-serving/charter.md:11]: Yes please!

File: wg-serving/charter.md
Comment (soltysh) [reply wg-serving/charter.md:34]: I agree with Clayton, I'd drop this point entirely.

File: wg-serving/charter.md
Comment (soltysh) [inline wg-serving/charter.md:87]: I'd like to see clearer exit criteria, maybe something along the deliverables? Iow. after achieving them and the items identified in the roadmap?

File: wg-serving/charter.md
Comment (mrbobbytables) [reply wg-serving/charter.md:11]: +1 it overlaps a bit with the batch charter as written.

**Review by mrbobbytables — CHANGES_REQUESTED** [review]

File: OWNERS_ALIASES
Comment (ArangoGutierrez) [inline OWNERS_ALIASES:145]:
```suggestion
    - ArangoGutierrez
```

File: sigs.yaml
Comment (ArangoGutierrez) [inline sigs.yaml:3490]:
```suggestion
    - github: ArangoGutierrez
```

File: sigs.yaml
Comment (soltysh) [reply sigs.yaml:3498]: Double checked with Bob, Arrango is good to go, Terry is not, since he's not a k8s org member.

Comment (ArangoGutierrez) [reply sigs.yaml:3498]: Typo is ArangoGutierreZ not ArangoGutierreX :)

Comment (terrytangyuan) [reply sigs.yaml:3498]: Thanks for checking. I have updated my membership request description in https://github.com/kubernetes/org/issues/4904.

Comment (SergeyKanzhelev) [reply sigs.yaml:3498]: @mrbobbytables I hope the list of @terrytangyuan and the ongoing work on kubeflow and other sister projects are enough. We also appreciate @terrytangyuan commitment to help  k8s community to understand and drive improvement for WG Serving use cases.

File: sig-list.md
Comment (SergeyKanzhelev) [reply sig-list.md]: it is already resolved in the yesterday version. Sorry for not resolving the comment

Comment (SergeyKanzhelev) [reply sig-list.md]: It is AM

File: wg-serving/charter.md
Comment (SergeyKanzhelev) [reply wg-serving/charter.md:34]: removed

File: wg-serving/charter.md
Comment (SergeyKanzhelev) [reply wg-serving/charter.md:11]: Added one paragraph about it

File: wg-serving/charter.md
Comment (SergeyKanzhelev) [reply wg-serving/charter.md:90]: I was generally following https://github.com/kubernetes/community/blob/master/wg-batch/charter.md#timelines-and-disbanding.

We don't yet know exact deliverables and as it's said in text we may end up developing a new sub project or it may be an external plugin. It is something we will need to decide.

What specific points are you looking for in this text?

File: wg-serving/charter.md
Comment (smarterclayton) [inline wg-serving/charter.md:12]: Serving and inference aren't interchangeable - thinking if there is a wording that could work better (this is not suggested wording, just trying to lay out the framing):

> Scope: Discuss and enhance serving workloads on Kubernetes, specifically focused on hardware-accelerated AI/ML inference

> Because serving and batch are both broad categories (pods with restartPolicy=Always vs those with restartPolicy=Never|OnFailure), we focus on the novel challenges of compute-intensive online inference, leveraging non-fungible accelerators vs highly generic CPU, and the unique reliability, efficiency, and cost obstacles those workloads face.  Our solutions should where possible provide automation that benefits other serving workloads like web services or stateful databases, be usable as primitives by multiple ecosystem projects, and compose well into the workflows of those deploying models to production.

How does that read?

Comment (SergeyKanzhelev) [reply wg-serving/charter.md]: thank you! rephrased a little bit. WDYT?

File: wg-serving/charter.md
Comment (soltysh) [reply wg-serving/charter.md:99]: Let's try to sketch at least the general exit criteria, for example try adding something like that to the first paragraph:

```
Achieving the aforementioned deliverables, also mentioned in the `In Scope` section,
will allow us to decide when to disband this WG. 
```

File: wg-serving/README.md
Comment (soltysh) [inline wg-serving/README.md:41]: Discussed with the rest of steering, you can put my name here.

**Review by soltysh — APPROVED** [review]
lgtm, modulo the steering update and Terry org membership

**Comment by soltysh** [comment]
/hold
so it doesn't get accidentally merged

File: wg-serving/charter.md
Comment (ArangoGutierrez) [inline wg-serving/charter.md:25]:
```suggestion
Discuss and enhance serving workloads on Kubernetes, specifically focusing on hardware-accelerated AI/ML inference.

The working group will focus on the novel challenges of compute-intensive online inference. Scenarios solving use cases involving non-fungible accelerators will be prioritized over solutions against generic CPU. However, all improvements should, where possible, benefit other serving workloads like web services or stateful databases, be usable as primitives by multiple ecosystem projects, and compose well into the workflows of those deploying models to production.

The Working Group Batch has a similar scope. The difference in scope by a simplified definition is that the Serving WG will generally concentrate on the workloads where Pods are running with restartPolicy=Always, while WG Batch will generally be looking at Pods with the restartPolicy=OnFailure. There are edge cases to this definition, but it creates an easy enough framework to differentiate the scope of these two Working Groups.
```

Some suggestions to the text

File: sigs.yaml
Comment (mrbobbytables) [reply sigs.yaml]: Terry's updated his app and should be good to go 👍

File: wg-serving/charter.md
Comment (mrbobbytables) [inline wg-serving/charter.md:25]: I really like this improvement. The section on the job types makes it quite clear. 👍

File: wg-serving/charter.md
Comment (mrbobbytables) [reply wg-serving/charter.md:99]: +1 to @soltysh 's comment. We've had several WGs that have gone on for years making minimal progress with no real exit criteria >_< Even if its fairly plane like that, its still worth it to have written down.

**Review by mrbobbytables — COMMENTED** [review]
general lgtm :)
I'd like to see the exit criteria that @soltysh mentioned added (or something similar), and we can merge once @terrytangyuan added as an org member. 👍

File: wg-serving/charter.md
Comment (smarterclayton) [reply wg-serving/charter.md]: Looks good

File: wg-serving/charter.md
Comment (SergeyKanzhelev) [reply wg-serving/charter.md:25]: Applied

Comment (SergeyKanzhelev) [reply wg-serving/charter.md:25]: thanks!

File: wg-serving/charter.md
Comment (BenTheElder) [reply wg-serving/charter.md:96]: Yes, this has been a repeat problem, workgroups are supposed to be scoped and finite, let's try to outline what the exit criteria are now.

File: sigs.yaml
Comment (SergeyKanzhelev) [reply sigs.yaml]: Created https://github.com/kubernetes/org/pull/4917 to expedite things a bit

File: wg-serving/charter.md
Comment (SergeyKanzhelev) [reply wg-serving/charter.md:96]: added, makes sense. Thank you for clarifying concerns

**Comment by SergeyKanzhelev** [comment]
Based on all the comments, this PR seems to be ready to be approved: @kubernetes/steering-committee 

The list of PRs I have now:

- add membership: https://github.com/kubernetes/org/pull/4917
- @terrytangyuan will accept the invite
- this PR merge
- Slack channel: https://github.com/kubernetes/community/pull/7830
- Mailing list: https://github.com/kubernetes/k8s.io/pull/6758

Please let me know if I am missing some other org PRs for WG creation.

**Comment by SergeyKanzhelev** [comment]
> general lgtm :) I'd like to see the exit criteria that @soltysh mentioned added (or something similar), and we can merge once @terrytangyuan added as an org member. 👍

All addressed now.

@mrbobbytables can you please take another look?

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Comment by SergeyKanzhelev** [comment]
Steering votes counting:

- @BenTheElder:
- @justaugustus: 
- @mrbobbytables: 
- @pacoxu: 
- @palnabarun: 
- @pohly: 
- @soltysh: +1 [link](https://github.com/kubernetes/community/pull/7823#pullrequestreview-2024818247)

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

File: sigs.yaml
Comment (pacoxu) [inline sigs.yaml:3474]: We already have one in https://www.youtube.com/watch?v=Ri4rQNoRBko. Can we create a playlist for the wg?

**Review by pacoxu — APPROVED** [review]
/lgtm

File: wg-serving/charter.md
Comment (palnabarun) [inline wg-serving/charter.md:83]: nit
```suggestion
Additionally, the WG commits to:
```

File: wg-serving/charter.md
Comment (palnabarun) [inline wg-serving/charter.md:78]: nit
```suggestion
This WG adheres to the Roles and Organization Management outlined in [wg-governance]
```

File: wg-serving/charter.md
Comment (palnabarun) [inline wg-serving/charter.md:90]: nit:
```suggestion
As a first mandate, the WG will define a roadmap in the first quarter of operation.
```

File: wg-serving/charter.md
Comment (palnabarun) [inline wg-serving/charter.md:53]: nit
```suggestion
- SIG Apps as a primary SIG
```

**Review by palnabarun — APPROVED** [review]
Only nits here and there. They can be taken in a follow up too if needed.

/lgtm
(on the charter with the steering hat)

**Review by mrbobbytables — APPROVED** [review]

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve

**Comment by SergeyKanzhelev** [comment]
Thank you everybody reviewed this! If there are any additional feedback, please do not hesitate to reach out to me.

Quorum on steering votes was reached: https://github.com/kubernetes/community/pull/7823#issuecomment-2080158480

/unhold

File: sigs.yaml
Comment (pnbrown) [reply sigs.yaml:3521]: Here is the WG Serving playlist:
https://www.youtube.com/playlist?list=PL69nYSiGNLP30qNanabU75ayPK7OPNAAS

Comment (SergeyKanzhelev) [reply sigs.yaml:3521]: thx, added it here: https://github.com/kubernetes/community/pull/7862

**Pull request merged**
